keps/sig-node/2411-cri-container-log-rotation/README.md

```diff
@@ -0,0 +1,301 @@
+- Supporting container runtimes that run on a different virtual/physical machine from the kubelet
+
+## Proposal
+
+Graduate the CRIContainerLogRotation feature gate from Beta to Stable. The kubelet already decides the container log directory structure and passes that down to the container runtime. The container runtime then writes the container's logs to this location. This makes the kubelet the best candidate to manage the rotation of the container logs as it already know the log directory structure and has access to it. The CRILogRotation feature gate implementation adds a container log manager package, which manages and rotates the container logs. It also adds 2 flags to the kubelet that allows the user to configure the maximum size of each log file and the maximum number of log files to retain. These flags are **--container-log-max-size** and **--container-log-max-files**.
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a kubernetes user, I want to use a CRI container runtime and want the container logs to be managed and rotated by the kubelet.
+
+#### Story 2
+
+As a kubernetes user, I want to integrate a logging agent that aggregates the logs to a remote storage without having to further integrate with the CRI container runtime.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+- Loss of some logs during log rotation. There is an open issue on this with a suggested fix https://github.com/kubernetes/kubernetes/issues/64760
+
+## Design Details
+
+This implementation adds container log manager package, which the kubelet uses to manage and rotate the logs. The container log manager will only start up when the container runtime being used is one that communicates with the kubelet via the CRI i.e CRI-O, containerd, etc.
+
+The rotated logs are compressed with gzip. The latest rotated log is not compressed as a logging agent, such as fluentd, might still be reading it right after rotation. The kubelet periodically checks the amount of disk space being used by the container logs and rotates them if the max value has been reached. After the logs are rotated, the kubelet sends a signal to the container runtime to re-open the log file.
+
+The user can configure the maximum size of a log file and the maximum number of log files to retain with the **--container-log-max-size** and **--container-log-max-files** flags. The default values are **10Mi** for the max file size and **5** for the max number of log files.
+
+The kubelet exposes a consistent log directory structure with embedded metadata so that logging agents can integrate with it is. The logging agents can directly work with the kubelet without having to make any further integrations with the container runtime in use.
+
+### Test Plan
+
+- There are currently unit tests and node E2E integration tests for container log rotation
+- Get feedback on performance and stability of the CRI log format on other products aside from OpenShift and GKE.
```

Comment:
One update since I brought it up at SIG Node last time: GKE already used CRIContainerLogRotation long time ago, and there is no compatibility issue.